groups/ntc/ntcr/ntcr_streamsocket.t.cpp

```diff
@@ -1070,6 +1120,136 @@ void StreamSocketManager::run()
         }
     }
 
+    {
+        // validate RX & TX timestamps using metrics
+        bsl::vector<bsl::shared_ptr<ntci::Monitorable> > monitorables;
+        ntcm::MonitorableUtil::loadRegisteredObjects(&monitorables);
+        for (bsl::vector<bsl::shared_ptr<ntci::Monitorable> >::iterator it =
+                 monitorables.begin();
+             it != monitorables.end();
+             ++it)
+        {
+            bdld::ManagedDatum stats;
+            (*it)->getStats(&stats);
+            const bdld::Datum& d = stats.datum();
+            NTCCFG_TEST_EQ(d.type(), bdld::Datum::DataType::e_ARRAY);
+            bdld::DatumArrayRef statsArray = d.theArray();
+
+            const int baseSchedDelayIndex   = 90;
+            const int baseSentDelayIndex    = 95;
+            const int baseAckDelayIndex     = 100;
+            const int baseReceiveDelayIndex = 105;
+
+            const int countOffset = 0;
+            const int totalOffset = 1;
+            const int minOffset   = 2;
+            const int avgOffset   = 3;
+            const int maxOffset   = 4;
+            const int total       = maxOffset + 1;
+
+            /// due to multithreaded nature of the tests it's hard to predict
+            /// the esxact amount of TX timestamps received. The implementation
+            // of ntcr_datagramsocket does not timestamp any outgoing packet
+            /// until the first TX timestamp is received from the reactor
+            const double txTimestampsPercentage = 0.48;
```

Comment:
Yes I agree, I do not like it either. And there is a chance that this can become an unstable test. Unfortunately I did not come up with a better solution.